doc/whatsnew/fragments/9214.internal

```diff
@@ -0,0 +1,4 @@
+Bump `astroid` to 3.2.3. This resolves an `AssertionError` arising
```

Comment:
I think we decided to focus on what happens for pylint for 3.2.0

```suggestion
Fixes an `AssertionError` arising
```
 Should not be an fragment either but I can't suggest that.